### PR TITLE
Update Leaderboard.js

### PR DESCRIPTION
**Pull request by JorgeVidal29**
Made the leaderboard api work

**Description**
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Resolves #(replace parentheses and the stuff here with the issue number, if there isn't one, create one.) 

**Type of change**
Check options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Fix or feature that would cause existing functionality to not work as expected
- [ ] This change requires a update in the design document

**Steps to Verify Functionality**
Make a bulleted/numbered list of steps to verify this feature/bugfix. Include screenshots if necessary.

**Final Checklist:**
Go down the list and check them off.

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the design document (if applicable)
- [x] My changes generate no new warnings
- [x] Pulled updated master branch (if changed since branch creation) and corrected any conflicts, if any occur.

### CODE REVIEW

File: src/globalComponents/Leaderboard.js [hunk 1]
```diff
@@ -22,10 +22,47 @@ class Leaderboard extends React.Component<Props, State> {
...
+		});
+		responsePromise.then(
+			response => response.json().then(data => {
+				if (response.status !== 200) {
+					console.log(response.status);
+					console.log(data.msg);
+					console.log(data);
+				}
+				else {
+					let leaders = [];
```
Comment (blaiadam) [inline src/globalComponents/Leaderboard.js]: There is a cleaner way of doing this:

`this.setState({leaderNames: data.map(usernames => usernames.userName})`

**Review by blaiadam — APPROVED** [review]
Works well, thank you.

File: src/globalComponents/Leaderboard.js (hunk 1 above)
Comment (JorgeVidal29) [reply src/globalComponents/Leaderboard.js]: Thanks that was a lot cleaner

**Review by blaiadam — APPROVED** [review]
Looks good.

**Pull request merged**